library/artifactory_security.py

```diff
@@ -0,0 +1,570 @@
+    sec_config = module.params['sec_config']
+    username = module.params['username']
+    auth_password = module.params['auth_password']
+    auth_token = module.params['auth_token']
+    validate_certs = module.params['validate_certs']
+    client_cert = module.params['client_cert']
+    client_key = module.params['client_key']
+    force_basic_auth = module.params['force_basic_auth']
+    state = module.params['state']
+    sec_config_dict = module.params['sec_config_dict']
+
+    if sec_config:
+        # temporarily convert to dict for validation
+        sec_config = ast.literal_eval(sec_config)
+
+    fail_messages = []
+
+    fails = art_base.validate_config_params(sec_config, sec_config_dict,
+                                            'sec_config',
+                                            'sec_config_dict')
+    fail_messages.extend(fails)
+    fails = art_base.validate_top_level_params('name', module, sec_config,
+                                               sec_config_dict,
+                                               'sec_config',
+                                               'sec_config_dict')
+    fail_messages.extend(fails)
+    fails = art_base.validate_top_level_params('password', module, sec_config,
+                                               sec_config_dict,
+                                               'sec_config',
+                                               'sec_config_dict')
+    fail_messages.extend(fails)
+    fails = art_base.validate_top_level_params('email', module, sec_config,
+                                               sec_config_dict,
+                                               'sec_config',
+                                               'sec_config_dict')
+    fail_messages.extend(fails)
+    fails = art_base.validate_top_level_params('repositories', module,
+                                               sec_config,
+                                               sec_config_dict,
+                                               'sec_config',
+                                               'sec_config_dict')
+    fail_messages.extend(fails)
+
+    # Populate failure messages
+    failure_message = "".join(fail_messages)
+
+    # Conflicting config values should not be resolved
+    if failure_message:
+        module.fail_json(msg=failure_message, **result)
+
+    sec_dict = dict()
+    if module.params['name']:
+        sec_dict['name'] = module.params['name']
+    if module.params['password']:
+        sec_dict['password'] = module.params['password']
+    if module.params['email']:
+        sec_dict['email'] = module.params['email']
+    if module.params['repositories']:
+        sec_dict['repositories'] = module.params['repositories']
+    if sec_config:
+        sec_dict.update(sec_config)
+    if sec_config_dict:
+        sec_dict.update(sec_config_dict)
+    sec_config = json.dumps(sec_dict)
+
+    result['original_message'] = ("Perform state '%s' against group '%s' "
```

Comment:
This says group, but really it could be a user, group, or permission set.

Reply:
Yeah, it is supposed to say "target".